### PR TITLE
feat(asset): Add query type to asset plugin

### PR DESCRIPTION
**Pull request by Catalin-web**
# Pull Request

## 🤨 Rationale

US: [[UI][Grafana] Merge Asset plugin and Asset Calibration plugins](https://ni.visualstudio.com/DevCentral/_backlogs/backlog/ASW%20SystemLink%20Asset%20Mgmt/Features/?workitem=2891104)

## 👩‍💻 Implementation

Refactored asset plugin, preparing asset calibration migration.

## 🧪 Testing

Moved current unit tests+
calibration-dashboard-divizion

## ✅ Checklist
- [x] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

File: src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            placeholder="Select systems"
+            width={85}
+            value={query.minionIds.map(toOption) || []} // Add default value
+          />
+        </InlineField>
+        <FloatingError message={errorMsg} />
+    </div>
+  );
+}
+const tooltips = {
```
Comment (TigranVardanyan) [inline src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx:104]: const tooltips = {
  workspace: `The workspace where you want to search for the assets.`,
  system: `Filter assets by system.`
};

File: src/datasources/asset/AssetCoordonatorDataSource.test.ts [hunk 2]
```diff
@@ -7,14 +7,14 @@ import {
...
   setupDataSource,
 } from "test/fixtures";
-import { AssetDataSource } from "./AssetDataSource";
-import { AssetMetadataQuery } from "./types";
+import { AssetCoordonatorDataSource } from "./AssetCoordonatorDataSource";
+import { ListAssetsQuery, AssetQueryType } from "./types";
 import { AssetPresenceWithSystemConnectionModel, AssetsResponse } from "datasources/asset-common/types";
 
-let ds: AssetDataSource, backendSrv: MockProxy<BackendSrv>
+let ds: AssetCoordonatorDataSource, backendSrv: MockProxy<BackendSrv>
```
Comment (kkerezsi) [inline src/datasources/asset/AssetCoordonatorDataSource.test.ts:14]: I don't think it's necessary to add `Coordonator` everywhere in the name. I'd keep it simply AssetDataSource as it was in the beginning

File: src/datasources/asset/types.ts [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
-export interface AssetMetadataQuery extends DataQuery {
+export interface ListAssetsQuery extends DataQuery {
   workspace: string,
   minionIds: string[]
 }
 
+export interface CalibrationForecastQuery extends DataQuery {
+}
+
+export interface AssetSummaryQuery extends DataQuery {
```
Comment (kkerezsi) [inline src/datasources/asset/types.ts:12]: To keep this more organized, I'm thinking we can split this in 3 files:

ListAssets.types.ts
CalibrationForecastQuery.types.ts
AssetSummaryQuery.types.ts

What do you think ?

File: src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    } else {
+      handleQueryChange({ ...query, workspace: '' }, true);
+    }
+  };
+
+  const handleMinionIdChange = (items: Array<SelectableValue<string>>): void => {
+    if (items && !_.isEqual(query.minionIds, items)) {
+      handleQueryChange(
+        { ...query, minionIds: items.map(i => i.value!) },
+        // do not run query if minionIds not changed
```
Comment (kkerezsi) [inline src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx:52]: Is this comment in the right place ?

File: src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        <MultiSelect
+          isClearable
+          allowCreateWhileLoading
+          options={loadMinionIdOptions()}
+          isValidNewOption={isValidId}
+          onChange={handleMinionIdChange}
+          placeholder="Select systems"
+          width={85}
+          value={query.minionIds.map(toOption) || []} // Add default value
+        />
```
Comment (kkerezsi) [inline src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx:98]: Don't need this comment either

File: src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        { ...query, minionIds: items.map(i => i.value!) },
+        // do not run query if minionIds not changed
+        true
+      );
+    } else {
+      handleQueryChange({ ...query, minionIds: [] }, true);
+    }
+  };
+
+  const getVariableOptions = (): Array<SelectableValue<string>> => {
```
Comment (kkerezsi) [inline src/datasources/asset/components/editors/list-assets/ListAssetsEditor.tsx:60]: we can leave this as is for now but I'll reach out to Josh to ask about this. I'm not sure if we should also include the variables as values in the system dropdown.

This is most likely going to be impacted by the query builder anyways.

File: src/datasources/asset/types.ts (hunk 3 above)
Comment (kkerezsi) [reply src/datasources/asset/types.ts:12]: We can keep in types.ts defined only the common declarations (such as AssetQuery)

**Review by kkerezsi — APPROVED** [review]

**Pull request merged**